torch/_prims_common/wrappers.py

```diff
@@ -320,12 +319,13 @@ def _fn(*args, out=None, **kwargs):
             sig.empty,
             out_type,
         )
-        params = chain(sig.parameters.values(), (out_param,))
+        params = *sig.parameters.values(), out_param
+        params = sorted(params, key=lambda p: p.kind)
         _fn.__signature__ = inspect.Signature(  # type: ignore[attr-defined]
             parameters=params, return_annotation=return_type  # type: ignore[arg-type]
         )
 
-        _fn.__annotations__ = fn.__annotations__
+        _fn.__annotations__ = dict(getattr(fn, "__annotations__", {}))
```

Comment:
why getattr rather than `fn.__annotations__`?

Reply:
It turns out the aten functions don't have an `__annotations__` member!  I had a comment there, I took it out as too much noise, perhaps I should put it back in?

Reply:
Can we use the annotations from the decomposition?

Reply:
Done!